src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -232,15 +232,41 @@ Namespace Microsoft.CodeAnalysis.Semantics
+            ' nodes, and deduplicate based on that. As an example:
+            '
+            ' Dim x, y = 1
+            '
+            ' This is an error scenario, but if we just use the BoundLocalDeclaration.Syntax.Parent directly, without deduplicating,
+            ' we'll end up with two IVariableDeclarations that have the same syntax node. So, we group by VariableDeclaratorSyntax
+            ' to put x and y in the same IVariableDeclaration
+
+            Dim groupedDeclarations = declarations.GroupBy(Function(decl)
+                                                               If decl.Kind = BoundKind.LocalDeclaration Then
+                                                                   Debug.Assert(decl.Syntax.Parent.IsKind(SyntaxKind.VariableDeclarator))
+                                                                   Return decl.Syntax.Parent
+                                                               Else
+                                                                   Return decl.Syntax
+                                                               End If
+                                                           End Function)
+
             Dim builder = ArrayBuilder(Of IVariableDeclaration).GetInstance()
-            For Each base In declarations
-                If base.Kind = BoundKind.LocalDeclaration Then
-                    Dim declaration = DirectCast(base, BoundLocalDeclaration)
-                    builder.Add(OperationFactory.CreateVariableDeclaration(declaration.LocalSymbol, Create(declaration.InitializerOpt), _semanticModel, declaration.Syntax))
-                ElseIf base.Kind = BoundKind.AsNewLocalDeclarations Then
-                    Dim asNewDeclarations = DirectCast(base, BoundAsNewLocalDeclarations)
+            For Each declGroup In groupedDeclarations
+                If declGroup.First().Kind = BoundKind.AsNewLocalDeclarations Then
+                    ' There should only ever be 1 AsNewLocalDeclarations per syntax, as we use its syntax directly
+                    Debug.Assert(declGroup.Count() = 1)
+                    Dim asNewDeclarations = DirectCast(declGroup.First(), BoundAsNewLocalDeclarations)
                     Dim localSymbols = asNewDeclarations.LocalDeclarations.SelectAsArray(Of ILocalSymbol)(Function(declaration) declaration.LocalSymbol)
                     builder.Add(OperationFactory.CreateVariableDeclaration(localSymbols, Create(asNewDeclarations.Initializer), _semanticModel, asNewDeclarations.Syntax))
+                ElseIf declGroup.First().Kind = BoundKind.LocalDeclaration Then
+                    ' Only the last of the declarations can have an initializer.
+                    Dim localSymbols = declGroup.SelectAsArray(Of ILocalSymbol)(Function(decl) DirectCast(decl, BoundLocalDeclaration).LocalSymbol)
+                    Dim initializer = Create(DirectCast(declGroup.Last(), BoundLocalDeclaration).InitializerOpt)
```

Comment:
>Dim initializer = Create(DirectCast(declGroup.Last(), BoundLocalDeclaration).InitializerOpt) [](start = 20, length = 92)

This probably won't do the right thing for a declaration ```Dim x(10), y As Integer``` or ```Dim x%(10), y$(11)```. It looks like the grouping approach cannot be used for these scenarios.

Reply:
It feels like we should go back to the drawing board.

---
In reply to: [144438762](https://github.com/dotnet/roslyn/pull/22682#discussion_r144438762) [](ancestors = 144438762)

Reply:
Hmm, yeah, this doesn't appear to work correctly for this. @jinujoseph, I think we're going to have to put this PR on hold for now, since we need to have the initializers for both variables.

Reply:
>  I think we're going to have to put this PR on hold for now

In the meantime, consider submitting unit-tests for these scenarios and other new tests that were added in this PR.